lib/internal/mime.js

```diff
@@ -182,103 +209,80 @@ class MIMEParams {
     for (const { 0: key, 1: value } of this.#data) {
       const encoded = encode(value);
       // Ensure they are separated
-      if (ret.length) ret += ';';
+      if (ret.length > 0) ret += ';';
       ret += `${key}=${encoded}`;
     }
+
     return ret;
   }
 
   // Used to act as a friendly class to stringifying stuff
   // not meant to be exposed to users, could inject invalid values
   static parseParametersString(str, position, params) {
     const paramsMap = params.#data;
-    const endOfSource = SafeStringPrototypeSearch(
-      StringPrototypeSlice(str, position),
-      START_ENDING_WHITESPACE,
-    ) + position;
-    while (position < endOfSource) {
+    const endOfSource = str.length;
+    const marker = { position };
+    while (marker.position < endOfSource) {
       // Skip any whitespace before parameter
-      position += SafeStringPrototypeSearch(
-        StringPrototypeSlice(str, position),
-        END_BEGINNING_WHITESPACE,
-      );
-      // Read until ';' or '='
-      const afterParameterName = SafeStringPrototypeSearch(
-        StringPrototypeSlice(str, position),
-        EQUALS_SEMICOLON_OR_END,
-      ) + position;
-      const parameterString = toASCIILower(
-        StringPrototypeSlice(str, position, afterParameterName),
-      );
-      position = afterParameterName;
-      // If we found a terminating character
-      if (position < endOfSource) {
-        // Safe to use because we never do special actions for surrogate pairs
-        const char = StringPrototypeCharAt(str, position);
-        // Skip the terminating character
-        position += 1;
+      moveUntil(str, isNotWhiteSpace, marker)
+      const parameterName = moveUntil(str, isSemicolonOrEquals, marker, { toASCII: true });
+      // If we are at end of the string, it cannot have a value
+      if (marker.position > endOfSource) break;
+      else {
         // Ignore parameters without values
-        if (char === ';') {
+        if (str[marker.position] === ';') {
```

Comment:
I recommend removing all string comparisons `str.charCodeAt(marker.position) === ...`

Reply:
This should performance identically due to the generated code being identical (at least it should almost be identical). The lowering was implemented a few years back for single characters.

Reply:
I applied the suggestion of replacing string checks with integers by using `char codes`. So far I was able to perceive an improvement of ~10% on top of the current improvement. I'll need to fix linting and testing but we are on the good direction 🙂